scripts/azure_new_vm_size_provision/watcher-new-azure-vm-size.ps1

```diff
@@ -0,0 +1,84 @@
+﻿if ( $customSecretsFilePath )
+{
+	$secretsFile = $customSecretsFilePath
+	Write-Host "Using provided secrets file: $($secretsFile | Split-Path -Leaf)"
+}
+if  ($env:Azure_Secrets_File)
+{
+	$secretsFile = $env:Azure_Secrets_File
+	Write-Host "Using predefined secrets file: $($secretsFile | Split-Path -Leaf) in Jenkins Global Environments."
+}
+if ( $secretsFile -eq $null )
+{
+    Write-Host "ERROR: Azure Secrets file not found in Jenkins / user not provided -customSecretsFilePath" -ForegroundColor Red -BackgroundColor Black
+    exit 1
+}
+
+
+if ( Test-Path $secretsFile)
+{
+	Write-Host "$($secretsFile | Split-Path -Leaf) found."
+    $xmlSecrets = [xml](Get-Content $secretsFile)
+    .\AddAzureRmAccountFromSecretsFile.ps1 -customSecretsFilePath $secretsFile
+	$subscriptionID = $xmlSecrets.secrets.SubscriptionID
+}
+else
+{
+	Write-Host "$($secretsFile | Split-Path -Leaf) file is not added in Jenkins Global Environments OR it is not bound to 'Azure_Secrets_File' variable." -ForegroundColor Red -BackgroundColor Black
+	Write-Host "Aborting." -ForegroundColor Red -BackgroundColor Black
+	exit 1
+}
+
+$newFile = "Z:\Jenkins_Shared_Do_Not_Delete\userContent\shared\azure-vm-sizes.txt"
+$oldFile = "Z:\Jenkins_Shared_Do_Not_Delete\userContent\shared\azure-vm-sizes-base.txt"
+$fallback ="Z:\Jenkins_Shared_Do_Not_Delete\userContent\shared\azure-vm-sizes-fallback.txt"
```

Comment:
all of these constants should not be here, try to move them in the Jenkinsfile?

Also, why do you need 3 files?
You can read the old values from file, and write the new ones in the same file.

Reply:
Still need t old file to compare but removing the fallback